bin/import_slr88.py

```diff
@@ -0,0 +1,225 @@
+        # Excluding samples that failed on label validation
+        counter["invalid_label"] += 1
+    elif int(frames / SAMPLE_RATE * 1000 / 15 / 2) < len(str(label)):
+        # Excluding samples that are too short to fit the transcript
+        counter["too_short"] += 1
+    elif frames / SAMPLE_RATE > MAX_SECS:
+        # Excluding very long samples to keep a reasonable batch-size
+        counter["too_long"] += 1
+    else:
+        # This one is good - keep it for the target CSV
+        rows.append((wav_filename, file_size, label))
+        counter["imported_time"] += frames
+    counter["all"] += 1
+    counter["total_time"] += frames
+
+    return (counter, rows)
+
+
+def _maybe_convert_sets(target_dir, extracted_data):
+    extracted_dir = os.path.join(target_dir, extracted_data)
+    # override existing CSV with normalized one
+    try:
+        target_csv_template = os.path.join(
+            target_dir, "".join([ARCHIVE_DIR_NAME, "_{}.csv"])
+        )
+        if os.path.isfile(target_csv_template):
+            return
+        else:
+            raise FileNotFoundError
+    except FileNotFoundError:
+
+        all_files = glob(f"{extracted_dir}/{ARCHIVE_NAME_txt}/*.txt")
+
+        transcripts = {}
+        for tr in all_files:
+            with open(tr, "r") as tr_source:
+                transcript = tr_source.read()
+                audio = os.path.basename(tr).replace(".txt", ".wav")
+                transcripts[audio] = transcript
+                tr_source.close()
+
+        # Get audiofile path and transcript for each sentence
+        samples = []
+        for record in glob(f"{extracted_dir}/{ARCHIVE_NAME_wav}/*.wav"):
+            record_file = os.path.basename(record)
+            if record_file in transcripts.keys():
+                samples.append((record, transcripts[record_file]))
+
+        # Keep track of how many samples are good vs. problematic
+        counter = get_counter()
+        num_samples = len(samples)
+        rows = []
+
+        print("Importing WAV files...")
+        pool = Pool()
+        bar = progressbar.ProgressBar(max_value=num_samples, widgets=SIMPLE_BAR)
+        for i, processed in enumerate(pool.imap_unordered(one_sample, samples), start=1):
+            counter += processed[0]
+            rows += processed[1]
+            bar.update(i)
+        bar.update(num_samples)
+        pool.close()
+        pool.join()
+
+        target_csv_template = os.path.join(
+            target_dir, "".join([ARCHIVE_DIR_NAME, "_{}.csv"])
+        )
+
+        with open(
+            target_csv_template.format("train"), "w", encoding="utf-8", newline=""
+        ) as train_csv_file:  # 80%
+            with open(
+                target_csv_template.format("dev"), "w", encoding="utf-8", newline=""
+            ) as dev_csv_file:  # 10%
+                with open(
+                    target_csv_template.format("test"), "w", encoding="utf-8", newline=""
+                ) as test_csv_file:  # 10%
```

Comment:
nit: you can open multiple context managers with comma:
```suggestion
        with open(
            target_csv_template.format("train"), "w", encoding="utf-8", newline=""  # 80%
        ) as train_csv_file, open(
            target_csv_template.format("dev"), "w", encoding="utf-8", newline=""  # 10%
        ) as dev_csv_file, open(
            target_csv_template.format("test"), "w", encoding="utf-8", newline=""  # 10%
        ) as test_csv_file:
```

Reply:
Yes much tidier indeed. Thanks for the tip.